File: backend/app/Support/SellerTypeField.php
<?php

namespace App\Support;

class SellerTypeField
{
    public const FIELD_ID = 'seller_type';

    public const OWNER_AR = 'مالك';

    public static function isField(array $field): bool
    {
        return ($field['id'] ?? null) === self::FIELD_ID;
    }

    /**
     * @return array{ar?: string, en?: string, tr?: string}|null
     */
    public static function findOwnerOption(array $field): ?array
    {
        foreach ($field['options'] ?? [] as $option) {
            if (is_array($option) && ($option['ar'] ?? null) === self::OWNER_AR) {
                return $option;
            }
            if (is_string($option) && $option === self::OWNER_AR) {
                return ['ar' => $option, 'en' => $option, 'tr' => $option];
            }
        }

        return null;
    }

    public static function ownerStoredValue(array $field, ?string $locale = null): string
    {
        $locale = $locale ?: app()->getLocale();
        $owner = self::findOwnerOption($field);
        if ($owner === null) {
            return self::OWNER_AR;
        }

        return (string) ($owner[$locale] ?? $owner['ar'] ?? self::OWNER_AR);
    }

    /**
     * Force seller_type to "مالك" when the user is not verified.
     *
     * @param  array<string, mixed>  $customFields
     * @param  array<int, array<string, mixed>>  $schema
     * @return array<string, mixed>
     */
    public static function applyLockedOwner(array $customFields, array $schema, $user, ?string $locale = null): array
    {
        if ($user && ($user->is_verified ?? false)) {
            return $customFields;
        }

        foreach ($schema as $field) {
            if (! self::isField($field)) {
                continue;
            }
            $customFields[self::FIELD_ID] = self::ownerStoredValue($field, $locale);

            break;
        }

        return $customFields;
    }
}

File: flutter/lib/helpers/seller_type_field.dart
/// صفة المعلن (seller_type): غير الموثق يُثبَّت على «مالك».
class SellerTypeField {
  SellerTypeField._();

  static const fieldId = 'seller_type';
  static const ownerAr = 'مالك';

  static bool isField(Map<String, dynamic> field) =>
      (field['id'] ?? '').toString() == fieldId;

  static Map<String, dynamic>? findOwnerOption(Map<String, dynamic> field) {
    final options = field['options'];
    if (options is! List) return null;
    for (final opt in options) {
      if (opt is Map) {
        final map = Map<String, dynamic>.from(opt);
        if ((map['ar'] ?? '').toString() == ownerAr) return map;
      } else if (opt?.toString() == ownerAr) {
        return {'ar': ownerAr, 'en': ownerAr, 'tr': ownerAr};
      }
    }
    return null;
  }

  /// قيمة التخزين كما في قوائم الاختيار (يفضّل ar للتوافق مع التطبيق).
  static String ownerStoredValue(
    Map<String, dynamic> field, {
    String Function(Map<String, dynamic> opt)? valueOf,
  }) {
    final owner = findOwnerOption(field);
    if (owner == null) return ownerAr;
    if (valueOf != null) {
      final v = valueOf(owner);
      if (v.isNotEmpty) return v;
    }
    return (owner['ar'] ?? owner['en'] ?? owner['tr'] ?? ownerAr).toString();
  }
}
